e2e/test/scenarios/embedding/sdk-iframe-embedding-setup/embed-parameters.cy.spec.ts

```diff
@@ -0,0 +1,242 @@
+      navigateToEmbedOptionsStep({
+        experience: "dashboard",
+        resourceName: "Dashboard with Parameters",
+      });
+
+      getEmbedSidebar().within(() => {
+        cy.findByText("Parameters").should("be.visible");
+
+        cy.log("parameter inputs should be visible");
+        cy.findByLabelText("ID").should("be.visible");
+        cy.findByLabelText("Product ID").should("be.visible");
+
+        cy.log("parameters should be visible by default");
+
+        parameterVisibilityToggle("id").should(
+          "have.attr",
+          "data-hidden",
+          "false",
+        );
+
+        parameterVisibilityToggle("product_id").should(
+          "have.attr",
+          "data-hidden",
+          "false",
+        );
+      });
+    });
+
+    it("can set default parameter values", () => {
+      navigateToEmbedOptionsStep({
+        experience: "dashboard",
+        resourceName: "Dashboard with Parameters",
+      });
+
+      cy.log("set default value for id");
+      getEmbedSidebar().findByLabelText("ID").type("123").blur();
+      H.getIframeBody()
+        .findByTestId("dashboard-parameters-widget-container")
+        .findByLabelText("ID")
+        .should("contain", "123");
+
+      cy.log("set default value for product id");
+      getEmbedSidebar().findByLabelText("Product ID").type("456").blur();
+      H.getIframeBody()
+        .findByTestId("dashboard-parameters-widget-container")
+        .findByLabelText("Product ID")
+        .should("contain", "456");
+
+      cy.log("both default values should be in the code snippet");
+      getEmbedSidebar().within(() => {
+        cy.findByText("Get Code").click();
+        codeBlock().should("contain", '"initialParameters"');
+        codeBlock().should("not.contain", '"hiddenParameters"');
+        codeBlock().should("contain", '"id": "123"');
+        codeBlock().should("contain", '"product_id": "456"');
+      });
+    });
+
+    it("can hide dashboard parameters", () => {
+      navigateToEmbedOptionsStep({
+        experience: "dashboard",
+        resourceName: "Dashboard with Parameters",
+      });
+
+      cy.log("hide both parameters");
+      getEmbedSidebar().within(() => {
+        parameterVisibilityToggle("id").click();
+        parameterVisibilityToggle("product_id").click();
+      });
+
+      cy.log("parameter widget container should not exist");
+      H.getIframeBody()
+        .findByTestId("dashboard-parameters-widget-container")
+        .should("not.exist");
+
+      cy.log("code snippet should contain the hidden parameters");
+      getEmbedSidebar().within(() => {
+        cy.findByText("Get Code").click();
+        codeBlock().should("contain", '"hiddenParameters"');
+        codeBlock().should("not.contain", '"initialParameters"');
+        codeBlock().should("contain", '"id"');
+        codeBlock().should("contain", '"product_id"');
+      });
+    });
+  });
+
+  it("shows no parameters message for dashboards without parameters", () => {
```

Comment:
Hmm, the describe block introduction isn't quite what I expected. I mean you don't have to only have a new describe when there's `beforeEach` inside do you?

We could have just describes for grouping related tests together. I see you structure the describe block as "dashboard with params"

You could have 3 more describes that way
- dashboards without params
- questions with params
- questions without params

Or you could do a less describes
- dashboards (includes both with and without params)
- questions

My point of having them in describe blocks is for pure organization, and making them easier to read and understand. Having some inside a describe block and some outside makes it much harder to understand which test are in which dimension.

Reply:
@WiNloSt fixed in https://github.com/metabase/metabase/pull/60303/commits/391fafe140343a002d151a83144a370487731ce7 in the downstream PR, I organized it into 3 categories:

- questions with parameters
- dashboards with parameters
- resources without parameters

much better now!